tests/test_orchestration_e2e.py

```diff
@@ -268,3 +268,132 @@ def orchestrator(ctx: task.OrchestrationContext, input: int):
         assert state.serialized_output == json.dumps(all_results)
         assert state.serialized_input == json.dumps(4)
         assert all_results == [1, 2, 3, 4, 5]
+
+def test_retry_policies():
+    # This test verifies that the retry policies are working as expected.
+    # It does this by creating an orchestration that calls a sub-orchestrator,
+    # which in turn calls an activity that always fails. 
+    # In the first setup, this setup is done without any retry policies, and
+    # the orchestration should fail. Here, sub-orchestrator is called twice,
+    # and the activity is called once.
+    # In the second setup, the retry policies are added, and the orchestration
+    # should still fail. But, number of times the sub-orchestrator and activity
+    # is called should increase as per the retry policies.
+
+    # First setup: No retry policies
```

Comment:
It seems weird to execute a test as part of `test_retry_policies` that doesn't have a retry policy. I suggest we break these into two separate tests, one which tests the default failure behavior and one which tests the behavior of retry policies.

Reply:
I have removed first setup i.e. without retry.
Other existing tests are supposed to cater / are catering to without retry cases.